src/domain/users/user-organizations.repository.integration.spec.ts

```diff
@@ -0,0 +1,1414 @@
+          orgId,
+        }),
+      ).rejects.toThrow('Organization not found.');
+    });
+  });
+
+  describe('updateRole', () => {
+    it('should update the role of a user organization', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const orgName = faker.word.noun();
+      const owner = await dbUserRepo.insert({
+        status: 'ACTIVE',
+      });
+      await dbWalletRepo.insert({
+        user: owner.generatedMaps[0],
+        address: authPayloadDto.signer_address,
+      });
+      const member = await dbUserRepo.insert({
+        status: 'ACTIVE',
+      });
+      const memberUserId = member.generatedMaps[0].id;
+      await dbWalletRepo.insert({
+        user: member.generatedMaps[0],
+        address: getAddress(faker.finance.ethereumAddress()),
+      });
+      const org = await dbOrgRepo.insert({
+        name: orgName,
+        status: 'ACTIVE',
+      });
+      const orgId = org.generatedMaps[0].id;
+      await dbUserOrgRepo.insert({
+        user: owner.generatedMaps[0],
+        organization: org.generatedMaps[0],
+        role: 'ADMIN',
+        status: 'ACTIVE',
+      });
+      await dbUserOrgRepo.insert({
+        user: member.generatedMaps[0],
+        organization: org.generatedMaps[0],
+        role: 'MEMBER',
+        status: 'ACTIVE',
+      });
+
+      await expect(
+        userOrgRepo.updateRole({
+          authPayload: new AuthPayload(authPayloadDto),
+          orgId,
+          role: 'ADMIN',
+          userId: memberUserId,
+        }),
+      ).resolves.not.toThrow();
+    });
+
+    it('should throw an error if the signer_address does not exist', async () => {
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('signer_address', undefined as unknown as `0x${string}`)
+        .build();
+      const orgId = faker.number.int({
+        min: 69420,
+        max: DB_MAX_SAFE_INTEGER,
+      });
+      const userId = faker.number.int({
+        min: 69420,
+        max: DB_MAX_SAFE_INTEGER,
+      });
+
+      await expect(
+        userOrgRepo.updateRole({
+          authPayload: new AuthPayload(authPayloadDto),
+          orgId,
+          role: 'ADMIN',
+          userId: userId,
+        }),
+      ).rejects.toThrow('Signer address not provided.');
+    });
+
+    it('should throw if the signer_address has no user', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const orgId = faker.number.int({
+        min: 69420,
+        max: DB_MAX_SAFE_INTEGER,
+      });
+      const userId = faker.number.int({
+        min: 69420,
+        max: DB_MAX_SAFE_INTEGER,
+      });
+
+      await expect(
+        userOrgRepo.updateRole({
+          authPayload: new AuthPayload(authPayloadDto),
+          orgId,
+          role: 'ADMIN',
+          userId,
+        }),
+      ).rejects.toThrow('User not found.');
+    });
+
+    it('should throw an error if the organization has no ACTIVE ADMINs', async () => {
```

Comment:
```suggestion
    it('should throw an error if user does not have access to upgrade to ADMIN', async () => {
```
An organization cannot exist without an admin, so I would test the scenarios that currently occur in our system.

Reply:
Changed in d556d8e14bac3e535f72e269f55bb37a7031d792.